data-raw/create_admiral_adsl.R

```diff
@@ -0,0 +1,79 @@
+#  Create dataset:   data/admiral_adsl.rda
+#  This script:  create_admiral_adsl.R creates dataset data/admiral_adsl.rda.
+#
+
+# styler:style_pkg()
+# devtools::lint()
+# devtools::spellcheck()
+
+# Preliminary
+library(diffdf)
+
+
+# To clarify directories (can be removed)
+CACHE_DIR <- "~/.cache/R/admiral_templates_data/"
+DATA_DIR <- "data-dir"
+DATA_RAW <- "data-raw"
+TEMPLATE_DIR <- "inst/templates/"
+
+# clean CACHE_DIR
+THE_FILE <- paste0(CACHE_DIR, "/adlb.rda")
+THE_FILE <- paste0(CACHE_DIR, "/adsl.rda")
+if (file.exists(THE_FILE)) file.remove(THE_FILE)
+
+#
+# STEPS
+#
+# First, use template to create the R script (in data-raw/admiral_adlb.R).
+# Next, source this script and create the data (~/.cache/R/admiral_template_data/admiral_adlb.rda)
+# Finally, shorten this data (now ~ 1.2 MB) by selecting only certain USUBJID
+
+# OMIT -- orignal method  - OMIT
+if (FALSE) {
+  # ### original mehtod (method 1)
+  # # First,  create the R script (from a template)
+  adam_name <- "adlb"
+  save_path <- paste0("./data-raw/admiral_", adam_name, ".R")
+  use_ad_template(
+    adam_name = adam_name,
+    save_path = save_path,
+    open = FALSE,
+    overwrite = TRUE
+  )
+  # Second, source the script and save data in .cache
+  source("data-raw/admiral_adlb.R") # nolint
+  load("~/.cache/R/admiral_templates_data/adlb.rda")
+}
+
+#
+#
+# Instead, USE template, as recommened by Buzz
+#
+source(paste0(TEMPLATE_DIR, "/ad_adsl.R")) # nolint
+load(paste0(CACHE_DIR, "adsl.rda"))
+admiral_adsl <- adsl
+
+#
+# Finally, save reduced ds
+#
+use_data(admiral_adsl, overwrite = TRUE)
+
+#
+#  TEST - is dataset identical to .... backup of unaltered dataset
+#
+e1 <- new.env()
+e2 <- new.env()
+load("data/admiral_adsl.rda", e1)
+
+# CHANGE to YOUR location of original dataset
+load("data-backup/admiral_adsl.rda", e2)
+
+
+identical(e1$admiral_adsl, e2$admiral_adsl)
+diffdf(e1$admiral_adsl, e2$admiral_adsl)
```

Comment:
Please specify the `key` argument.

Reply:
**ADSL**

BASE= backed copy
KEYS = ?   Correct?

```
capture.output(diffdf(compare=e1$admiral_adsl, base=e2$admiral_adsl, keys  = c("STUDYID", "USUBJID")),
               file="data-raw/diffdf_23SEPT")
```

```
Differences found between the objects!

A summary is given below.

There are columns in BASE and COMPARE with different modes !!
All rows are shown in table below

  ================================
   VARIABLE  MODE.BASE  MODE.COMP 
  --------------------------------
    AGEGR1    numeric   character 
  --------------------------------

There are columns in BASE and COMPARE with different classes !!
All rows are shown in table below

  ==================================
   VARIABLE  CLASS.BASE  CLASS.COMP 
  ----------------------------------
    AGEGR1     factor    character  
  ----------------------------------

There are columns in COMPARE that are not in BASE !!
All rows are shown in table below

  =========
   COLUMNS 
  ---------
   DTHDTF  
   DTHCAUS 
   DTHDOM  
   DTHCGR1 
  ---------

Not all Values Compared Equal
All rows are shown in table below

  =============================
   Variable  No of Differences 
  -----------------------------
   LSTALVDT          9         
  -----------------------------


All rows are shown in table below

  =============================================================
   VARIABLE    STUDYID       USUBJID       BASE      COMPARE   
  -------------------------------------------------------------
   LSTALVDT  CDISCPILOT01  01-703-1258  2013-01-11  2013-01-25 
   LSTALVDT  CDISCPILOT01  01-704-1009  2013-09-25  2013-09-27 
   LSTALVDT  CDISCPILOT01  01-704-1017  2013-11-18  2013-11-22 
   LSTALVDT  CDISCPILOT01  01-704-1266  2013-12-06  2013-12-23 
   LSTALVDT  CDISCPILOT01  01-704-1332  2014-02-16  2014-02-24 
   LSTALVDT  CDISCPILOT01  01-708-1216  2012-12-06  2012-12-13 
   LSTALVDT  CDISCPILOT01  01-710-1070  2013-02-05  2013-02-10 
   LSTALVDT  CDISCPILOT01  01-710-1271  2012-11-22  2012-11-26 
   LSTALVDT  CDISCPILOT01  01-718-1427  2013-02-18  2013-02-25 
  -------------------------------------------------------------


```

Reply:
Yes, `keys  = c("STUDYID", "USUBJID")` is correct for ADSL.

Reply:
@bundfussr how did we get these variables into ADSL?  

Oh maybe from the vignette?   Maybe this ADSL was created from the vignette?

![image](https://github.com/user-attachments/assets/c65a1e62-6cfb-47b2-a62e-24d249dbeaea)

@jimrothstein could you use the purl option for Rmarkdown to pull all the code out of the adsl vignette and see what you get?

`knitr::purl("adsl.Rmd", output = "output_file.R")`